custom_components/google_home/api.py

```diff
@@ -309,6 +325,49 @@ async def reboot_google_device(self, device: GoogleHomeDevice) -> None:
                 device.name,
             )
 
+    async def get_or_set_do_not_disturb(
+        self, device: GoogleHomeDevice, enable: Optional[bool] = None
+    ) -> GoogleHomeDevice:
+        """Gets or sets the do not disturb setting on a Google Home device."""
+
+        data = None
+
+        if enable is None:
+            _LOGGER.debug(
+                "Getting Do Not Disturb setting from Google Home device %s",
+                device.name,
+            )
+        else:
+            data = {"notifications_enabled": not enable}
+            _LOGGER.debug(
+                "Setting Do Not Disturb setting to %s on Google Home device %s",
+                not enable,
+                device.name,
+            )
+
+        response = await self.post(
+            endpoint=API_ENDPOINT_DO_NOT_DISTURB, data=json.dumps(data), device=device
```

Comment:
I think you shouldn't call `json.dumps` at all.
```python
>>> json.dumps(None)
'null'
```

Typing may be difficult in this case. In `types.py` you can define such type:
```python
JsonDict = Dict[str, Union[bool, int, str, List[str]]
```
It's not the exact type JSON dict may have, but it's enough for all these cases.

And then use `Optional[JsonDict]` for `data` arg of `post()`.

Reply:
Thanks, I tried early on not using `json.dumps` and had problems with `types`. Will test this. 😄

Reply:
I get this error with mypy:

```text
custom_components/google_home/api.py:277: error: Argument "data" to "post" of "GlocaltokensApiClient" has incompatible type "Dict[str, List[str]]"; expected "Optional[Dict[str, Union[bool, int, str, List[str]]]]"
custom_components/google_home/api.py:277: note: "Dict" is invariant -- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance
custom_components/google_home/api.py:277: note: Consider using "Mapping" instead, which is covariant in the value type
custom_components/google_home/api.py:319: error: Argument "data" to "post" of "GlocaltokensApiClient" has incompatible type "Dict[str, str]"; expected "Optional[Dict[str, Union[bool, int, str, List[str]]]]"
custom_components/google_home/api.py:319: note: "Dict" is invariant -- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance
custom_components/google_home/api.py:319: note: Consider using "Mapping" instead, which is covariant in the value type
custom_components/google_home/api.py:350: error: Argument "data" to "post" of "GlocaltokensApiClient" has incompatible type "Optional[Dict[str, bool]]"; expected "Optional[Dict[str, Union[bool, int, str, List[str]]]]"
```

Reply:
Do what it suggests:
```python
JsonDict = Mapping[str, Union[bool, int, str, List[str]]
```